pkg/testutils/kyma.go

```diff
@@ -0,0 +1,134 @@
+package testutils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"github.com/kyma-project/lifecycle-manager/pkg/watcher"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	ErrStatusModuleStateMismatch = errors.New("status.modules.state not match")
+	ErrKymaNotDeleted            = errors.New("kyma CR not deleted")
+)
+
+func NewTestKyma(name string) *v1beta2.Kyma {
+	return newKCPKymaWithNamespace(name, v1.NamespaceDefault, v1beta2.DefaultChannel, v1beta2.SyncStrategyLocalClient)
+}
+
+func NewKymaForE2E(name, namespace, channel string) *v1beta2.Kyma {
```

Comment:
Can we move this to the e2e tests when it's only used by e2e tests?

Reply:
I actually encourage reuse of these util functions for e2e test, many of them can be shared, like check state, conditions, we don't need to strictly separate them.

Reply:
Of course, but they can be part of e2e_test/utils then right?

Reply:
My idea is to use `testutils` as a shared package between all tests, if not necessary, do not need to differentiate with e2e test and rest.

Of course, if some specific function is only suitable for e2e, then putting it under the `e2e_test` package makes sense. But it's a bit overkill for kyma initialization. We only have kyma different between e2e and unit test, because of `SyncStrategy`. But for rest resources, the initialization of test resources can be shared.

And follow the naming pattern, actually, it's better to put it under `e2e_test/kyma.go`, it's the only use case, I just think it's not necessary to have this level of separation, at least at the moment. If you want, we can have a call.

Reply:
Just not a big fan of offering functions by pkg for other tests if the specific test scenarios need a Kyma with sync label, where they can then just use the builder and create a Kyma with that label explicitly, explaining via code what is so special about this "e2e Kyma", because one can notice then "ah it is just a Kyma with a sync label" without needing to jump to a definition in another file. And if a convenience function is **really** needed to save that one line of code, then it should in my opinion reside as close to the usage as possible and not in some generic testutils pkg. 
But at this point I don't really care anymore. So maybe either you can rename it to `NewKymaWithSyncLabel` to explain that this is e2e agnostic, remove it or leave it. 👍

Reply:
renamed to `NewKymaWithSyncLabel`